pkg/controller/migmigration/task.go

```diff
@@ -1397,7 +1432,7 @@ func (t *Task) anyFlags(phase Phase) (bool, error) {
 	if phase.any&IndirectImage != 0 && t.indirectImageMigration() {
 		return true, nil
 	}
-	if phase.any&EnableImage != 0 && !t.PlanResources.MigPlan.IsImageMigrationDisabled() {
+	if phase.any&EnableImage != 0 && !t.PlanResources.MigPlan.IsImageMigrationDisabled() && t.migrateState() {
 		return false, nil
```

Comment:
This looks like a pre-existing bug, but the above should return `true`. The "all" test returns false when any included test fails. This is the "any" test, which returns true any time an included test succeeds. Here, the "EnableImage" test finds that image migration is *not* disabled, so it should return true. Now, for this PR, my question is whether an `EnableImage` itinerary check should also return true for stage migration? I think the answer is "no" -- so this line should be "true" and at 1435 we should *not* be adding `t.migrateState()` here.

Reply:
I think the same true/false bug exists below for EnableVolume.

Reply:
@sseago one thing I don't understand is that how was it working until now?

Reply:
@pranavgaikwad I think the bug wasn't exposed anywhere because it looks like we're not actually using the "any" test with EnableImage anywhere. It's mainly used with "all".